### PR TITLE
[dynamo] Extend LazyVariableTracker to tuples

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117426

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by lezcano — APPROVED** [review]
Good catch! We should audit a bit the uses of `as_python_constant` as we may have quite a few of these.

File: torch/_dynamo/variables/builtin.py [hunk 1]
```diff
@@ -487,6 +487,13 @@ def unwrap_unspec_args_kwargs(args, kwargs):
             k: v.as_python_constant() for k, v in kwargs.items()
         }
 
+    def get_constant_handler(self, args, kwargs):
```
[the remark this answers is not in the document]
Comment (anijain2305) [reply torch/_dynamo/variables/builtin.py]: Yes, thanks for catching it.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**